src/gen-ts.ts

```diff
@@ -130,70 +134,40 @@ function handleElementOrBehavior(
     return;
   }
 
-  const extends_ = new Array<string>();
-  if (isPolymerElement(feature)) {
-    // TODO Only do this when it's a legacy Polymer element.
-    extends_.push('Polymer.Element');
-    for (const behavior of feature.behaviorAssignments) {
-      extends_.push(behavior.name);
-    }
-  } else if (isElement(feature)) {
-    extends_.push('HTMLElement');
-  }
-
-  const properties = new Array<ts.Property>();
-  for (const property of feature.properties.values()) {
-    if (property.inheritedFrom) {
-      continue;
-    }
-    properties.push({
-      kind: 'property',
-      name: property.name || '',
-      description: property.description || '',
-      type: property.type ? closureTypeToTypeScript(property.type) : ts.anyType,
+  if (isClassBased) {
+    parent.members.push({
```

Comment:
add comments saying what and why these branches are doing

Reply:
Hopefully more clear now with the new variables and comment above.